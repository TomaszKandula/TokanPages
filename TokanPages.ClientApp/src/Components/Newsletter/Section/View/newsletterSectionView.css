.newsletter-margins {
    margin: 3rem;
}

@media (max-width: 768px) {
    .newsletter-margins {
        margin: 1.5rem;
    }
}
