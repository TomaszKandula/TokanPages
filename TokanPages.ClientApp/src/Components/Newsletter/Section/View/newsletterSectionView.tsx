import * as React from "react";
import { ViewProperties } from "../../../../Shared/Abstractions";
import { ReactChangeEvent, ReactKeyboardEvent } from "../../../../Shared/types";
import { Animated, ProgressBar, Skeleton, TextField } from "../../../../Shared/Components";

interface NewsletterViewProps extends ViewProperties {
    isMobile: boolean;
    caption: string;
    text: string;
    keyHandler: (event: ReactKeyboardEvent) => void;
    formHandler: (event: ReactChangeEvent) => void;
    email: string;
    buttonHandler: () => void;
    progress: boolean;
    buttonText: string;
    labelEmail: string;
    background?: string;
}

const ActiveButton = (props: NewsletterViewProps): React.ReactElement => (
        <button
            type="submit"
            onClick={props.buttonHandler}
            className="bulma-button bulma-is-link bulma-is-light bulma-is-fullwidth"
            disabled={props.progress}
        >
            {!props.progress ? props.buttonText : <ProgressBar size={20} />}
        </button>
    );

export const NewsletterSectionView = (props: NewsletterViewProps): React.ReactElement => (
        <section className={props.background}>
            <div className="bulma-container">
                <div className="py-6">
                    <div className={props.isMobile ? "p-5" : "p-6"}>
                        <div className="bulma-columns bulma-is-vcentered">
                            <div className="bulma-column">
                                <Animated dataAos="fade-down" dataAosDelay={150}>
                                    <Skeleton isLoading={props.isLoading} mode="Text" height={32}>
                                    <p className="is-size-3 has-text-grey-dark has-text-centered">{props.caption}</p>
                                    </Skeleton>
                                </Animated>
                                <Animated dataAos="zoom-in" dataAosDelay={200}>
                                    <Skeleton isLoading={props.isLoading} mode="Text" height={24}>
                                    <p className="is-size-5 has-text-grey has-text-centered">{props.text}</p>
                                    </Skeleton>
                                </Animated>
                            </div>
                            <div className="bulma-column">
                                <Animated dataAos="zoom-in" dataAosDelay={300}>
                                    <Skeleton isLoading={props.isLoading} mode="Rect" height={40}>
                                    <TextField
                                        required
                                        uuid="email"
                                        autoComplete="email"
                                        onKeyUp={props.keyHandler}
                                        onChange={props.formHandler}
                                        value={props.email}
                                        placeholder={props.labelEmail}
                                    />
                                    </Skeleton>
                                </Animated>
                            </div>
                            <div className="bulma-column">
                                <Animated dataAos="zoom-in" dataAosDelay={350}>
                                    <Skeleton isLoading={props.isLoading} mode="Rect" height={40}>
                                    <ActiveButton {...props} />
                                    </Skeleton>
                                </Animated>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </section>
    );
