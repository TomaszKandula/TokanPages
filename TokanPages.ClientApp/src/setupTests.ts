// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
