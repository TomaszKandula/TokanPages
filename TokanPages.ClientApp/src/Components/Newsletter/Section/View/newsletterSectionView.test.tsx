import "../../../../setupTests";
import React from "react";
import { render } from "@testing-library/react";
import { NewsletterSectionView } from "./newsletterSectionView";

describe("test component: newsletterView", () => {
    it("should render correctly '<NewsletterView />' when content is loaded.", () => {
        const html = render(
            <NewsletterSectionView
                isLoading={false}
                caption={"Join the newsletter!"}
                text={"We will never share your email address."}
                keyHandler={jest.fn()}
                formHandler={jest.fn()}
                email={"ester.exposito@gmail.com"}
                buttonHandler={jest.fn()}
                progress={false}
                buttonText={"Subscribe"}
                labelEmail={"Email address"}
            />
        );

        expect(html).toMatchSnapshot();
    });
});
