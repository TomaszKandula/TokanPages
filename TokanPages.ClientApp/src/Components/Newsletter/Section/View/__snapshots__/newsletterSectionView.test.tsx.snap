// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: newsletterView should render correctly '<NewsletterView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="bulma-container"
        >
          <div
            class="py-6"
          >
            <div
              class="p-6"
            >
              <div
                class="bulma-columns bulma-is-vcentered"
              >
                <div
                  class="bulma-column"
                >
                  <div
                    data-aos="fade-down"
                    data-aos-delay="150"
                  >
                    <p
                      class="is-size-3 has-text-grey-dark has-text-centered"
                    >
                      Join the newsletter!
                    </p>
                  </div>
                  <div
                    data-aos="zoom-in"
                    data-aos-delay="200"
                  >
                    <p
                      class="is-size-5 has-text-grey has-text-centered"
                    >
                      We will never share your email address.
                    </p>
                  </div>
                </div>
                <div
                  class="bulma-column"
                >
                  <div
                    data-aos="zoom-in"
                    data-aos-delay="300"
                  >
                    <div
                      class="bulma-control  "
                    >
                      <input
                        autocomplete="email"
                        class="bulma-input   "
                        id="email"
                        name="email"
                        placeholder="Email address"
                        required=""
                        type="text"
                        value="ester.exposito@gmail.com"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="bulma-column"
                >
                  <div
                    data-aos="zoom-in"
                    data-aos-delay="350"
                  >
                    <button
                      class="bulma-button bulma-is-link bulma-is-light bulma-is-fullwidth"
                      type="submit"
                    >
                      Subscribe
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="bulma-container"
      >
        <div
          class="py-6"
        >
          <div
            class="p-6"
          >
            <div
              class="bulma-columns bulma-is-vcentered"
            >
              <div
                class="bulma-column"
              >
                <div
                  data-aos="fade-down"
                  data-aos-delay="150"
                >
                  <p
                    class="is-size-3 has-text-grey-dark has-text-centered"
                  >
                    Join the newsletter!
                  </p>
                </div>
                <div
                  data-aos="zoom-in"
                  data-aos-delay="200"
                >
                  <p
                    class="is-size-5 has-text-grey has-text-centered"
                  >
                    We will never share your email address.
                  </p>
                </div>
              </div>
              <div
                class="bulma-column"
              >
                <div
                  data-aos="zoom-in"
                  data-aos-delay="300"
                >
                  <div
                    class="bulma-control  "
                  >
                    <input
                      autocomplete="email"
                      class="bulma-input   "
                      id="email"
                      name="email"
                      placeholder="Email address"
                      required=""
                      type="text"
                      value="ester.exposito@gmail.com"
                    />
                  </div>
                </div>
              </div>
              <div
                class="bulma-column"
              >
                <div
                  data-aos="zoom-in"
                  data-aos-delay="350"
                >
                  <button
                    class="bulma-button bulma-is-link bulma-is-light bulma-is-fullwidth"
                    type="submit"
                  >
                    Subscribe
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
